### ISSUE TITLE
Refactor frontend before starting project

### ISSUE THREAD

**Issue by sueszli** [sueszli/matchy#10]
**Most importantly: Fix inefficient / buggy design for user state**
Currently, when logging the user in or registering them we are observing 2 states at once:
- The state of `user` (based on `supabase.auth.onAuthStateChange`, ran in `App.vue`)
- The state of the user `profile` (based on `user.ts`)

**This causes 2 problems:**
- These two states depend on eachother and cause some highly inconsistent behaviour when logging in / registering.
Ideally we would want to have a single `profile` state, containing all relevant data, that the app reacts to.
- Another issue is that `profile` and `user` are based on different tables from the backend that we fetch individually and merge in the frontend.

**Solution:**
- Creating a [postgres view](https://www.postgresql.org/docs/9.2/sql-createview.html) that lets us fetch all relevant user data at once.
- Reducing the number of states in the frontend. This requires a redesign (I'm unsure how it could be done).

**Comment by stefnotch** [comment]
The user should probably be treated as `isLoggedIn` and mostly everything else should land in the `profile`

**Comment by stefnotch** [comment]
Oh yeah, I tweaked the user auth code a little bit. Now mostly everything is in the store, and thus it should be less terrible.

(I'm currently taking a look at how to make our error-handling bullet-proof)

**Comment by sueszli** [comment]
Reminder to myself: Fix some inconsistencies regarding the naming

We differentiate between participants and organizers, not visitors and organizers
